### PR TITLE
PR 64

### PR DESCRIPTION
**Pull request by ericsorenson**
### Bugs

- [7974 BUG: Legacy eAccess flag is missing from petitioner contact info ](https://github.com/flexion/ef-cms/issues/7974) 
- [7923 BUG: Updating a New Trial Session Proceeding Type is Not Saved](https://github.com/flexion/ef-cms/issues/7923) 
- [7920 BUG: IRS practitioner sees incorrect service preference when address is sealed ](https://github.com/flexion/ef-cms/issues/7920) 
- [7919 BUG: Logged out when trying to access trial session ](https://github.com/flexion/ef-cms/issues/7919) 
- [7916 BUG: Private Practitioner Can Search For Sealed Case By Petitioner Name](https://github.com/flexion/ef-cms/issues/7916) 
- [7910 BUG: Logged out when trying to view document QC processed tab](https://github.com/flexion/ef-cms/issues/7910) 
- [7908 BUG: Petitions Clerk Filing Fee "Waived" date validation bug](https://github.com/flexion/ef-cms/issues/7908) 
- [7907 BUG: Signature/Served stamp overlay issue](https://github.com/flexion/ef-cms/issues/7907) 
- [7872 BUG: Closed case added to trial session not displaying session info on case ](https://github.com/flexion/ef-cms/issues/7872) 
- [7827 BUG: Signature not staying affixed to Proposed Stipulated Decision](https://github.com/flexion/ef-cms/issues/7827) 
- [7819 BUG: Unable to edit migrated Order (code "O") due to validation error ](https://github.com/flexion/ef-cms/issues/7819) 
- [7772 BUG: Transcript should be viewable after 90 days](https://github.com/flexion/ef-cms/issues/7772) 
- [7679 BUG: Typing a case detail URL w/ suffix or w/ a docket # that doesn't exist blank page](https://github.com/flexion/ef-cms/issues/7679) 
- [7678 BUG: New coversheet not being applied consistently when docket entry is updated](https://github.com/flexion/ef-cms/issues/7678) 
- [7590 BUG: Notice of Docket Change not generated with Attachment and C/S info changes](https://github.com/flexion/ef-cms/issues/7590) 
- [7572 BUG: Document QC: Field to concatenate document title is not populated on QC screen ](https://github.com/flexion/ef-cms/issues/7572) 
- [7510 BUG: Edit Docket Entry: Docket entry preview not updating with date for scenario D documents ](https://github.com/flexion/ef-cms/issues/7510) 

Refactoring Tasks
- https://trello.com/c/AUeI9m8G
- https://trello.com/c/X4RGd1pj
- https://trello.com/c/IG9Y3PDa
- https://trello.com/c/cEDKY5xy
- https://trello.com/c/BlC808XK
- https://trello.com/c/ANfWIIfv
- https://trello.com/c/LbrYJtpz
- https://trello.com/c/TxNIb1Sl
- https://trello.com/c/Qnch6jhC
- https://trello.com/c/9M6CrqM2
- https://trello.com/c/z4uNMRQx

### Migrations

- None

### Deployment Notes
- Terraform upgraded to 13.6
- NodeJS upgraded to 14.16.0 will require rebuilding the ECR image

### CODE REVIEW

**Comment by klohman** [comment]
## PR 64 Release Notes

### Internal Only 

**Bug Fix** Waived date form error on Petition QC is removed when error is corrected

**Bug Fix** Practitioners can no longer see unassociated sealed cases when searching by petitioner name.

**Bug Fix** Users can access a case detail screen directly by the URL with or without the docket number suffix.

**Bug Fix** If a closed case is added to a trial session, the session information now displays accurately on the case.

**Bug Fix** To prevent service stamps and signatures from overlapping, signatures can no longer be placed where stamps are applied.  

**Bug Fix** Updates to trial session proceeding type are now accurately saved.

**Bug Fix** Docket clerk can now update docket entries for migrated event code “O” documents.

**Bug Fix** A Notice of Docket Change is now generated when Attachment(s) and / or Certificate of Service information is added, removed, or updated during QC.  

**Bug Fix** Users are no longer logged out when accessing Los Angeles, Hybrid trial sessions. Data limits were being reached when retrieving eligible cases. We have removed redundant data to remain below the limit.  

**Bug Fix** Users are no longer logged out when accessing the Section Document QC Processed tab. Data limits were being reached when attempting to view these documents. We are now retrieving less data per document.

**Bug Fix** Petitioners with eAccess in old system now display the eAccess flag.

**Bug Fix** Signatures are now consistently saved and visible on Proposed Stipulated Decisions.  

**Bug Fix** A new cover sheet is consistently applied when updating docket entry document type after QC.

**Bug Fix** Transcripts with trial/hearing date more than 90 days ago are visible to parties. Transcripts migrated from old system are visible 90 days after their filing date.

**Bug Fix** Docket entry preview now displays accurate date and description when editing Certificate of Service or Amended Certificate of Service.


### External / Public
**Bug Fix** Petitioner service preference displays accurately for parties when address is sealed.

**Bug Fix** Transcripts with trial/hearing date more than 90 days ago are visible to parties. Transcripts migrated from old system are visible to parties 90 days after their filing date.

File: shared/src/business/utilities/createSignedDocumentsForInspection.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+const fs = require('fs');
+const path = require('path');
+const {
+  applicationContext: applicationContext,
+} = require('../../../../web-client/src/applicationContext');
+const {
+  generateSignedDocumentInteractor,
+} = require('../useCases/generateSignedDocumentInteractor');
+
+// see this for reference: shared/src/business/utilities/documentGenerators.test.js
```
Comment (mmarcotte) [inline shared/src/business/utilities/createSignedDocumentsForInspection.js:17]: Could we add a bit more about the purpose of this script? Use this to do X?

File: shared/src/persistence/dynamodbClientService.js [hunk 2]
```diff
@@ -278,56 +278,41 @@ exports.batchGet = async ({ applicationContext, keys }) => {
...
-    return Promise.resolve();
+  const results = batchDeleteItems(items);
+
+  if (!isEmpty(results.UnprocessedItems)) {
+    const retryResults = batchDeleteItems(results.UnprocessedItems);
+
+    if (!isEmpty(retryResults.UnprocessedItems)) {
+      applicationContext.logger.error(
+        'Unable to batch delete',
+        JSON.stringify(retryResults.UnprocessedItems, null, 2),
```
Comment (mmarcotte) [inline shared/src/persistence/dynamodbClientService.js]: I think the second parameter for this function expects an object not a string. has this been tested?

File: web-client/integration-tests/docketClerkAddsTranscriptToDocketRecord.test.js [hunk 3]
```diff
@@ -21,6 +18,7 @@ describe('Docket Clerk Adds Transcript to Docket Record', () => {
   const { TRANSCRIPT_EVENT_CODE } = applicationContext.getConstants();
 
   beforeAll(() => {
+    console.error = () => {};
```
Comment (mmarcotte) [inline web-client/integration-tests/docketClerkAddsTranscriptToDocketRecord.test.js:21]: can this be removed?

File: web-client/src/presenter/actions/navigation/toggleMenuStateAction.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
+// eslint-disable-next-line spellcheck/spell-checker
```
Comment (mmarcotte) [inline web-client/src/presenter/actions/navigation/toggleMenuStateAction.js]: might be better to just change `DropdownMenu` => `DropDownMenu`. And hyphenate unsetting if it still complains

File: shared/src/business/utilities/createSignedDocumentsForInspection.js (hunk 1 above)
Comment (ericsorenson) [reply shared/src/business/utilities/createSignedDocumentsForInspection.js:17]: You bet!

**Comment by mmarcotte** [comment]
Any idea why the Sonar report hasn't been run since January 28th?
https://sonarcloud.io/organizations/flexion-github/projects

**Comment by mmarcotte** [comment]
Also looks like the ZAP report isn't updating any more either

File: web-client/src/presenter/actions/navigation/toggleMenuStateAction.js (hunk 4 above)
Comment (ericsorenson) [reply web-client/src/presenter/actions/navigation/toggleMenuStateAction.js]: Updated!

File: web-client/integration-tests/docketClerkAddsTranscriptToDocketRecord.test.js (hunk 3 above)
Comment (ericsorenson) [reply web-client/integration-tests/docketClerkAddsTranscriptToDocketRecord.test.js:21]: This was added to keep the console clean during the test execution.

File: shared/src/persistence/dynamodbClientService.js (hunk 2 above)
Comment (ericsorenson) [reply shared/src/persistence/dynamodbClientService.js]: Updated!

File: shared/src/business/utilities/createSignedDocumentsForInspection.js (hunk 1 above)
Comment (ericsorenson) [reply shared/src/business/utilities/createSignedDocumentsForInspection.js:17]: Updated!

**Comment by ericsorenson** [comment]
Sonarcloud is still running on all branches https://sonarcloud.io/dashboard?branch=develop&id=ef-cms-api

**Review by mmarcotte — APPROVED** [review]
Great Work!

Great work, thanks!

**Pull request merged**